clients/rospy/src/rospy/timer.py

```diff
@@ -59,6 +59,21 @@ def __init__(self, hz):
         self.last_time = rospy.rostime.get_rostime()
         self.sleep_dur = rospy.rostime.Duration(0, int(1e9/hz))
 
+    def remaining(self):
```

Comment:
This just a suggestion (feel free not to implement it), but this method could take an optional `curr_time` argument, so that you could reuse curr_time in line 73 and be slightly more accurate. E.g.:

```python
def remaining(self, curr_time_time=None):
    if curr_time is None:
        curr_time = rospy.rostime.get_rostime()
    # detect time jumping backwards
    if self.last_time > curr_time:
        self.last_time = curr_time

    # calculate remaining time
    elapsed = curr_time - self.last_time
    return self.sleep_dur - elapsed
```

and then you would call this inside `sleep` like this:

```python
    curr_time = rospy.rostime.get_rostime()
    sleep(self.remaining(curr_time=curr_time))
    self.last_time = self.last_time + self.sleep_dur
```